daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,157 @@
+    exit 0
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the operating system
+
+OS_DETECTION=
+NAME=
+VERSION=
+VERSION_ID=
+ID=
+ID_LIKE=
+
+if [ -f "/etc/os-release" ]; then
+    OS_DETECTION="/etc/os-release"
+    eval "$(grep -E "^(NAME|ID|ID_LIKE|VERSION|VERSION_ID)=" </etc/os-release)"
+elif [ -f "/etc/lsb-release" ]; then
+    OS_DETECTION="/etc/lsb-release"
+    DISTRIB_ID=
+    DISTRIB_RELEASE=
+    DISTRIB_CODENAME=
+    eval "$(grep -E "^(DISTRIB_ID|DISTRIB_RELEASE|DISTRIB_CODENAME)=" </etc/lsb-release)"
+    NAME="${DISTRIB_ID}"
+    VERSION="${DISTRIB_RELEASE}"
+    ID="${DISTRIB_CODENAME}"
+elif [ ! -z "$(command -v lsb_release 2>/dev/null)" ]; then
+    OS_DETECTION="lsb_release"
+    NAME="$(lsb_release -i 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION_ID="$(lsb_release -f 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION="$(lsb_release -c 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the kernel
+
+KERNEL_NAME="$(uname -s)"
+KERNEL_VERSION="$(uname -r)"
+ARCHITECTURE="$(uname -m)"
+
+# -------------------------------------------------------------------------------------------------
+# detect the virtualization
+
+VIRTUALIZATION="unknown"
+VIRT_DETECTION="none"
+
+if [ ! -z "$(command -v systemd-detect-virt 2>/dev/null)" ]
+then
+    VIRTUALIZATION="$(systemd-detect-virt)"
+    VIRT_DETECTION="systemd-detect-virt"
+else
+    # /proc/1/sched exposes the host's pid of our init !
+    # http://stackoverflow.com/a/37016302
+    pid=$( head -n 1 </proc/1/sched 2>/dev/null | { IFS='(),#:' read -r _unused pid _unused _unused; echo $pid; } )
+    if [ ! -z "${pid}" ]; then
+        pid=$(( pid + 0 ))
+        [ ${pid} -ne 1 ] &&
+            VIRTUALIZATION="container" &&
+            VIRT_DETECTION="pid"
+    fi
+
+    # ubuntu and debian supply /bin/running-in-container
+    # https://www.apt-browse.org/browse/ubuntu/trusty/main/i386/upstart/1.12.1-0ubuntu4/file/bin/running-in-container
+    [ -x "/bin/running-in-container" ] &&
+        "/bin/running-in-container" >/dev/null 2>&1 &&
+        VIRTUALIZATION="container" &&
+        VIRT_DETECTION="/bin/running-in-container"
+
+    # lxc sets environment variable 'container'
+    [ ! -z "${container}" ] &&
+        VIRTUALIZATION="lxc" &&
+        VIRT_DETECTION="containerenv"
+
+    # docker creates /.dockerenv
+    # http://stackoverflow.com/a/25518345
+    [ -f "/.dockerenv" ] &&
+        VIRTUALIZATION="docker" &&
+        VIRT_DETECTION="dockerenv"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# check netdata version
+
+if [ -z "${NETDATA_VERSION}" ]; then
+    NETDATA_VERSION="$(netdata -V 2>&1 | cut -d ' ' -f 2)"
```

Comment:
What is the execution scenario where `NETDATA_VERION` is prepopulated?

Are you sure that `$(netdata -V 2>&1 | cut -d ' ' -f 2)` will be executed when netdata was installed in a non-default installation path (when `NETDATA_PREFIX` was used during installation)?

Reply:
When you install in non-default location, then it is entirely possible that command `netdata` ends up in location not included in `$PATH`. For example assume installation in `/home/user/apps`.

Reply:
Made it like this:
```
if [ -z "${NETDATA_VERSION}" ]; then
	NETDATA_VERSION="uknown"
	netdata -V >/dev/null 2>&1 && NETDATA_VERSION="$(netdata -V 2>&1 | cut -d ' ' -f 2)"
fi
```

Reply:
This was fixed from your commit.